app/controllers/katello/api/v2/host_errata_controller.rb

```diff
@@ -61,6 +72,34 @@ def applicability
       respond_for_async :resource => task
     end
 
+    def find_bulk_errata_ids(bulk_params)
+      #works on a structure of param_group bulk_params and transforms it into a list of errata_ids
+      bulk_params[:included] ||= {}
+      bulk_params[:excluded] ||= {}
+      @errata = []
+
+      unless bulk_params[:included][:ids].blank?
+        @errata = @host.content_facet.installable_errata.where(:errata_id => bulk_params[:included][:ids])
+      end
+
+      if bulk_params[:included][:search]
+        search_errata = @host.content_facet.installable_errata
+        search_errata = search_errata.search_for(bulk_params[:included][:search])
+        if @errata.any?
+          ::Katello::Erratum.where("errata_id in (?) OR errata_id in (?)", @errata, search_errata)
```

Comment:
Couldn't you combine @errata and search_errata if they are both arrays and use `::Katello::Erratum.where(errata_id: combined_array)`?

There is also the `.or` active record method that would also work

Reply:
I like this idea. However there could be a 1000 errata ids matching a search criteria, using the query as above will give AR the opportunity to be smart and do something like ```errata_id in (select errata_id from .... where(..search..) )``` . As opposed to hydrating the 1000 ids in a query

Reply:
AR support OR now. Going to use that